File: Sources/CountriesFlagsObjC/source/UIImage+Ireland.m
//
//  UIImage+Ireland.m
//  CountriesFlags
//

#import "UIImage+Ireland.h"

@implementation UIImage (Ireland)

+(UIImage * _Nullable) flagIrelandWithSize:(CGSize)size {
    UIColor* colorGreen = [UIColor colorWithRed:22.0f/255.0f
                                           green:155.0f/255.0f
                                            blue:98.0f/255.0f
                                           alpha:1.0f];
    UIColor* colorWhite = [UIColor whiteColor];
    UIColor* colorRed = [UIColor colorWithRed:1.0f
                                        green:136.0f/255.0f
                                         blue:62.0f/255.0f
                                        alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorGreen setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorWhite setFill];
    UIRectFill(CGRectMake(size.width / 3.0, 0.0f, size.width / 3.0, size.height));
    
    [colorRed setFill];
    UIRectFill(CGRectMake(size.width / 3.0 * 2.0, 0.0f, size.width / 3.0, size.height));
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Ireland.swift
//
//  UIImage+Ireland.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Ireland "wikipedia"
     
     Generates the flag of Ireland.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Ireland.
     */
    internal static func flagIreland(with size: CGSize) -> UIImage? {
        let colorGreen = UIColor(red: 22.0/255.0,
                                 green: 155.0/255.0,
                                 blue: 98.0/255.0,
                                 alpha: 1.0)
        let colorWhite = UIColor.white
        let colorRed = UIColor(red: 1.0,
                               green: 136.0/255.0,
                               blue: 62.0/255.0,
                               alpha: 1.0)

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorGreen.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorWhite.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: size.width / 3.0, y: 0.0),
                          size: CGSize(width: size.width / 3.0, height: size.height)))

        colorRed.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: size.width / 3.0 * 2.0, y: 0.0),
                          size: CGSize(width: size.width / 3.0, height: size.height)))

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
